
<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta charset="utf-8" />
    <title>nats_dataprocess_neighbors module &#8212; Bioinfo Nats 0.0.1 documentation</title>
    <link rel="stylesheet" href="_static/alabaster.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <script type="text/javascript" src="_static/language_data.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
   
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <div class="section" id="module-nats_dataprocess_neighbors">
<span id="nats-dataprocess-neighbors-module"></span><h1>nats_dataprocess_neighbors module<a class="headerlink" href="#module-nats_dataprocess_neighbors" title="Permalink to this headline">¶</a></h1>
<p>Analysis for active sites neigbors in NATs</p>
<p>This script summarizes the energy score of specific residues in the available structures (relaxed pdb file)
in the current directory.
The structures scores are listed in the relaxed pdb files and the script will isolate only the resiudes neighbouring
the catalytic sites. The resulting score is summed over this selection.</p>
<p>The resulting data is logged in a csv file ‘pyrosetta_out.csv’ for further analyses.</p>
<dl class="function">
<dt id="nats_dataprocess_neighbors.neighbor_res_select">
<code class="sig-prename descclassname">nats_dataprocess_neighbors.</code><code class="sig-name descname">neighbor_res_select</code><span class="sig-paren">(</span><em class="sig-param">pdbfilename</em>, <em class="sig-param">sel_chain</em>, <em class="sig-param">cutoff</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/nats_dataprocess_neighbors.html#neighbor_res_select"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#nats_dataprocess_neighbors.neighbor_res_select" title="Permalink to this definition">¶</a></dt>
<dd><p>Selection of the catalytic residue neighbours from the structure pdb files.</p>
<p>It uses NeighborSearch method from BioPDB. All the detected residues indices are returned in a list.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>pdbfilename</strong> (<em>str</em>) – Pdb file name</p></li>
<li><p><strong>sel_chain</strong> (<em>str</em>) – Identification letter of the structure chain</p></li>
<li><p><strong>cutoff</strong> (<em>float</em>) – Cutoff value in Angstroms for neighbouring search</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>List of detected residue indices</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>list</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="nats_dataprocess_neighbors.read_msa_fasta">
<code class="sig-prename descclassname">nats_dataprocess_neighbors.</code><code class="sig-name descname">read_msa_fasta</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/nats_dataprocess_neighbors.html#read_msa_fasta"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#nats_dataprocess_neighbors.read_msa_fasta" title="Permalink to this definition">¶</a></dt>
<dd><p>Reads multiple structure alignment from MUSTANG.</p>
<p>It determines the structurally aligned core of the proteins.
Note: here, only the aligned regions are of interest, gaps are removed.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns</dt>
<dd class="field-odd"><p>Dictionary. Keys: structure pdb id, Values: aligned indices</p>
</dd>
<dt class="field-even">Return type</dt>
<dd class="field-even"><p>dict</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="nats_dataprocess_neighbors.read_pdb_catalytic">
<code class="sig-prename descclassname">nats_dataprocess_neighbors.</code><code class="sig-name descname">read_pdb_catalytic</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/nats_dataprocess_neighbors.html#read_pdb_catalytic"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#nats_dataprocess_neighbors.read_pdb_catalytic" title="Permalink to this definition">¶</a></dt>
<dd><p>Reads the list of catalytic residues listed in pdb_catalytic.txt file in ../data/input/etc</p>
<dl class="field-list simple">
<dt class="field-odd">Returns</dt>
<dd class="field-odd"><p>Dictionary. Keys: structure pdb id, Values: catalytic residue index</p>
</dd>
<dt class="field-even">Return type</dt>
<dd class="field-even"><p>dict</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="nats_dataprocess_neighbors.read_pdb_starts">
<code class="sig-prename descclassname">nats_dataprocess_neighbors.</code><code class="sig-name descname">read_pdb_starts</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/nats_dataprocess_neighbors.html#read_pdb_starts"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#nats_dataprocess_neighbors.read_pdb_starts" title="Permalink to this definition">¶</a></dt>
<dd><p>Reads at which index each pdb sequence is starting from the pdb_starts.txt file from ../data/input/etc</p>
<dl class="field-list simple">
<dt class="field-odd">Returns</dt>
<dd class="field-odd"><p>Dictionary. Keys: structure pdb id, Values: starting index</p>
</dd>
<dt class="field-even">Return type</dt>
<dd class="field-even"><p>dict</p>
</dd>
</dl>
</dd></dl>

</div>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<h1 class="logo"><a href="index.html">Bioinfo Nats</a></h1>








<h3>Navigation</h3>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="index.html">Documentation overview</a><ul>
  </ul></li>
</ul>
</div>
<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" />
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>








        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2020, Pierre Bedoucha.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 2.1.2</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.12</a>
      
      |
      <a href="_sources/nats_dataprocess_neighbors.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>